<launch>

    <rosparam file="$(find explorer_turtle)/param/ros.yaml"/>

    <arg name="markerSize"      default="0.1"/>    <!-- in m -->
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->

    <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera_tf" args="0 0 0 0 1.57 0 map camera 10" />

    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video1" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="camera" />
        <param name="io_method" value="mmap"/>
        <param name="camera_info_url" value="file://$(find explorer_turtle)/param/usb_cam.yaml"/> <!--switch for new camera-->
        <param name="camera_name" value="usb_cam"/><!--switch for new camera-->
        <param name="focus" value="10"/>
  </node>

    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
        <remap from="/camera_info" to="/usb_cam/camera_info" />
        <remap from="/image" to="/usb_cam/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="camera"/>
    </node>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false">
        <remap from="image" to="/aruco_marker_publisher/result"/>
        <param name="autosize" value="true" />
    </node>

</launch>
